plugins/flatpak/gs-flatpak.c

```diff
@@ -3144,9 +3215,27 @@ gs_flatpak_file_to_app_repo (GsFlatpak *self,
 {
 	g_autoptr(GsApp) app = NULL;
 	g_autoptr(FlatpakRemote) xremote = NULL;
+	GFileType file_type = G_FILE_TYPE_UNKNOWN;
 
 	/* create app */
-	app = gs_flatpak_create_app_from_repo_file (self, file, cancellable, error);
+	file_type = g_file_query_file_type (file, G_FILE_QUERY_INFO_NONE, cancellable);
+	if (file_type == G_FILE_TYPE_DIRECTORY) {
+		g_autoptr(GError) local_error = NULL;
+		app = gs_flatpak_create_app_from_repo_dir (self, file, cancellable,
+							   &local_error);
+		if (app == NULL) {
+			g_autofree gchar *path = g_file_get_path (file);
+			g_debug ("Failed to get app from dir %s: %s; skipping", path,
+				 local_error->message);
+			return TRUE;
```

Comment:
I added this on purpose because IIRC this would e.g. fail in the system instance of gs-flatpak and prevent the further processing by the user's gs-flatpak instance.
Maybe something more clever could be done but I was in a rush and decided to let it silently fail (the reasoning being: in comparison with the regular file-to-app logic, having an //.ostree// dir in a USB drive is a less strict action than having a //.flatpakref// file, so silently failing it not as critical).